### PR TITLE
github/workflows: remove arm64 jobs

### PR DESCRIPTION
**Pull request by ivanvc**
Remove the ARM64 jobs from GitHub workflows. Add a comment regarding the use of sudo in robustness tests.

Link to #848

### CODE REVIEW

**Comment by ivanvc** [comment]
@jmhbnz and I have been discussing the migration of robustness tests. The current issue is that we can't install `xfsprogs` in the container/prow job pod (https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/etcd-io_bbolt/844/pull-bbolt-robustness-arm64/1849591798716960768#). The reason is highly likely that the host node doesn't have the XFS filesystem. Therefore, we can't use dm-flakey.

As we will need help from the infra team, James suggests reaching out to `k/test-infra` and `#sig-k8s-infra`, which I'll do shortly.

**Comment by ivanvc** [comment]
@ahrtr, do you know (or provide some guidance on) why the 4 CPU tests with race are getting stuck? I could add a timeout, but that would make the job fail. I'm running them in Prow using the same command we used in the GitHub workflow.

Refer to https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/etcd-io_bbolt/846/pull-bbolt-test-4-cpu-race-arm64/1851126914400391168

**Comment by ahrtr** [comment]
> do you know (or provide some guidance on) why the 4 CPU tests with race are getting stuck?

It looks like the test infra's issue, the test process somehow terminated or being blocked. 1000 concurrent  goroutines shouldn't cause too much resource pressure. Not sure whether we can request some help from test-infra, probably they have more insight.

> === RUN   TestSimulateNoFreeListSync_10000op_1000p
>     btesting.go:47: Opening bbolt DB at: /tmp/TestSimulateNoFreeListSync_10000op_1000p788848248/001/db
>

**Comment by ivanvc** [comment]
I'll mark this PR ready for review to check the triggered prow jobs.

File: Makefile [hunk 1]
```diff
@@ -90,7 +90,8 @@ test-failpoint:
 	@echo "[failpoint] array freelist test"
 	BBOLT_VERIFY=all TEST_FREELIST_TYPE=array go test -v ${TESTFLAGS} -timeout 30m ./tests/failpoint
 
-.PHONY: test-robustness # Running robustness tests requires root permission
+.PHONY: test-robustness # Running robustness tests requires root permission for now
+# TODO: Remove sudo once we fully migrate to the prow infrastructure
```
Comment (ahrtr) [inline Makefile:94]: Question on the TODO comment: won't removing sudo on prow break the robustness test, since it requires root permission?

Comment (ivanvc) [reply Makefile:94]: Prow jobs already run as the super user, and `sudo` is not installed. When we migrated etcd jobs, we had to remove `sudo` calls. (cc. @jmhbnz)

**Comment by ivanvc** [comment]
@ahrtr, other than the comment on the `TODO`. The jobs triggered look fine, and they are all green.

**Review by ahrtr — APPROVED** [review]
Thanks

**Pull request merged**